libandroid-navigation/src/main/java/com/mapbox/services/android/navigation/v5/navigation/NavigationService.java

```diff
@@ -279,39 +290,30 @@ private void rerouteSessionsStateUpdate() {
     recentDistancesFromManeuverInMeters.clear();
     mapboxNavigation.getEventDispatcher().onUserOffRoute(rawLocation);
     mapboxNavigation.setSessionState(
-      mapboxNavigation.getSessionState().toBuilder().lastReroutePosition(
-        Position.fromLngLat(rawLocation.getLongitude(), rawLocation.getLatitude())).build()
-    );
+      mapboxNavigation.getSessionState().toBuilder().lastRerouteLocation(rawLocation).build());
   }
 
   public void rerouteOccurred() {
     mapboxNavigation.setSessionState(mapboxNavigation.getSessionState().toBuilder()
+      .routeProgressBeforeReroute(routeProgress)
       .beforeRerouteLocations(Arrays.asList(
         locationBuffer.toArray(new Location[locationBuffer.size()])))
-      .routeProgressBeforeReroute(routeProgress)
+      .lastRerouteDate(new Date())
```

Comment:
Won't this cause `SessionState.timeFromLastReroute()` to take the diff from the current reroute date rather than the previous reroute date?